src/main/java/com/github/boscojared/disparse/parser/Parser.java

```diff
@@ -1,28 +1,34 @@
-package disparse.parser;
+package com.github.boscojared.disparse.parser;
+
+import com.github.boscojared.disparse.parser.exceptions.NoCommandNameFound;
+import com.github.boscojared.disparse.parser.exceptions.OptionRequiresValue;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
 
-import disparse.parser.exceptions.NoCommandNameFound;
-import disparse.parser.exceptions.OptionRequiresValue;
-import java.util.*;
-import java.util.stream.Collectors;
 
 public class Parser {
-    private Map<Command, ? extends Collection<Flag>> commandToFlags;
-    private Map<String, Flag> longOptionMap = new HashMap<>();
-    private Map<Character, Flag> shortOptionMap = new HashMap<>();
+    private final Map<Command, ? extends Collection<CommandFlag>> commandToFlags;
+    private final Map<String, CommandFlag> longOptionMap = new HashMap<>();
+    private final Map<Character, CommandFlag> shortOptionMap = new HashMap<>();
 
-    public Parser(final Map<Command, ? extends Collection<Flag>> commandToFlags) {
+    public Parser(final Map<Command, ? extends Collection<CommandFlag>> commandToFlags) {
         this.commandToFlags = commandToFlags;
     }
 
     public ParsedOutput parse(List<String> args) {
         final Command commandName = this.findCommand(args);
 
-        this.createShortAndLongOptions(commandName);
+        fillShortAndLongOptions(commandName);
```

Comment:
Just for consistency use `this.fillShortAndLongOptions` I pretty much always use `this` in this situation, and having it mixed-and-matched all over won't be great

Reply:
right, i never seen this used for methods in constructors